pkg/admission/pod_eviction_test.go

```diff
@@ -0,0 +1,576 @@
+package admission
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/grafana/rollout-operator/pkg/config"
+	"github.com/grafana/rollout-operator/pkg/zpdb"
+)
+
+const (
+	statefulSetZoneA  = "ingester-zone-a"
+	statefulSetZoneB  = "ingester-zone-b"
+	statefulSetZoneC  = "ingester-zone-c"
+	testPodZoneA0     = "ingester-zone-a-0"
+	testPodZoneA1     = "ingester-zone-a-1"
+	testPodZoneA2     = "ingester-zone-a-2"
+	testPodZoneB0     = "ingester-zone-b-0"
+	testPodZoneB1     = "ingester-zone-b-1"
+	testPodZoneB2     = "ingester-zone-b-2"
+	testPodZoneC0     = "ingester-zone-c-0"
+	testPodZoneC1     = "ingester-zone-c-1"
+	testPodZoneC2     = "ingester-zone-c-2"
+	testNamespace     = "test-dev-0"
+	rolloutGroupValue = "ingester"
+	rolloutGroupLabel = config.RolloutGroupLabelKey
```

Comment:
Do we need our own constant here if `config.RolloutGroupLabelKey` already exists?